### ISSUE TITLE
Hardcode Level BBS

### ISSUE THREAD

**Issue by GoogleCodeExporter** [pennmush/pennmush#898]
```
Is this a command or @function?

Command

What will the permission restrictions be?
Configurable.

What will this do?
It would standardize what is nearly a universal softcode system.


Can this be done in softcode?
No. It cannot be done in softcode. 

----
<Public> Chicken Boo Wiggles says, "BBS might be worthwhile to make into a 
standard feature maybe. Considering how nearly universal Mydrian BBS is"
<Public> Really Cheetah, Mike says, "(As far as Myrddin's goes, I'd be more 
inclined to add a proper hardcoded BBsystem than bundle it with Penn, even if 
we were able to do the latter (I don't know what the license is, off the top of 
my head).)"
<Public> Cool Points: 4 Ghost says, "that would be a great idea"
<Public> Really Cheetah, Mike says, "(However, I have about 0.3 inclination to 
bundle Myrddin's BB system, and about 2.6 inclination to hardcode a version. 
These are out of 100. So don't hold your breath.;)"
<Public> Really Cheetah, Mike says, "(Actually, if I'm honest, it's more like a 
26/100, I'm just kinda lazy and apathetic most of the time;)"
<Public> Chicken Boo Wiggles says, "But you can add in can_dark"
<Public> Cool Points: 4 Ghost laughs
<Public> Really Cheetah, Mike is pretty sure that wasn't him.
<Hardcode> Mike says, "Is there seriously interest for a bundled BB system?"
<Hardcode> Ashen-Shugar says, "not sure it's worth the effort"
<Hardcode> Ghost says, "think about it, what is nearly the first thing that 
someone asks for after they get help bringing up a new server?"
<Hardcode> Ghost says, "What is the link to Myrddin's BBS?"
<Hardcode> Mike says, "myrddin.annelions.com I think. GIYF."
<Hardcode> Ghost says, "just saying."
<Hardcode> Anarchist Orange Wiggles says, "Um, well in my experience Myrddin 
BBS is almost always the first thing to get put on the mu*"
<Hardcode> Ghost says, "when was the last time you added a major feature to 
Penn?"
<Hardcode> Anarchist Orange Wiggles says, "256 colors"
Hardcode> Ashen-Shugar says, "would the hardcode buy anything over the existing 
softcode to warrent making it?"
<Hardcode> Mike says, "Myrddin's system isn't necessarily installed because 
it's /good/, though, it's installed because it's pretty much all there is, 
which has kinda made it an industry standard. People who need a BB and can't 
write one ask, and everyone says "Myrddin's"."
<Hardcode> Anarchist Orange Wiggles says, "Myrddin's is pretty functional."
<Hardcode> Mike says, "Me personally, Ghost, or that one was added to Penn? 
Also depends what you consider a 'major feature'."
<Hardcode> Ashen-Shugar says, "the reason to hardcode mail was it was horible 
for cleanup and handling large databases.  the bbs system doesn't suffer that 
really."
<Hardcode> Ghost tries reverse psychology.  "You probably can't write one 
better anyway."
<Hardcode> Ashen-Shugar grins.
<Hardcode> Anarchist Orange Wiggles says, "Wasnt Page made a standard feature 
due to its universal nature?"
<Hardcode> Mike says, "Standardisation is also a useful reason, though, AS - 
basically every game has a BB system, and it's one of the things - like 
channels and mail - that it's useful for everyone to know going into a game."
<Hardcode> Ashen-Shugar says, "true Mike, but sometimes softcode should remain 
softcode, myrddin is kinda a fence sitter."
<Hardcode> Mike would also like to see a hardcoded system built into @mail.
<Hardcode> Ghost says, "Thats two yes's and a maybe"


```

Original issue reported on code.google.com by `AgentWig...@gmail.com` on 25 Sep 2013 at 2:22

**Comment by tkrajcar** [comment]
I think this would have been a good idea at one time, but now I think it might be too late. How many **new** games are starting that would actually use a hardcode BBS? Given the effort to write one of these, thoroughly debug it etc, AND build the proper flexibility into things like permissioning, etc., vs the tiny effort needed to get a Myrddin BBS that has all of those things...

**Comment by talvo** [comment]
There are a bunch of changes/improvements that `@mail` really needs anyway, so doing it all together wouldn't be unreasonable.

**Comment by HarryCordewener** [comment]
Not to mention that customizing a hardcode BBS is a pain in the rear after the fact. I vote against this. Fixing @mail however, I am all for!
